src/derived/histogram.jl

```diff
@@ -1,19 +1,72 @@
 function histogram(; x::AbstractVector = Int[],
-					relativefreq::Bool = false,
-					horizontal::Bool = false,
-					nbins::Int = 10)
+                    relativefreq::Bool = false,
+                    horizontal::Bool = false,
+                    nbins::Int = 10,
+                    bandwidth::Real=0.0)
 
-	a, b = hist(x, nbins)
-	total = relativefreq == false? 1 : sum(b)
+    if bandwidth == 0.0
+        a, b = hist(x, nbins)
+        bandwidth=a.step/a.divisor
+    elseif bandwidth < 0.0
+		error("bandwidth should be a positive number but got $(bandwidth)")
+	else
+        xmin, xmax = extrema(x)
+        a, b = hist(x, xmin:bandwidth:xmax)
+    end
+    total = relativefreq ? sum(b)*bandwidth : 1
+
+    #v = barplot(x = a[2:end], y = b/total)
+    v = VegaVisualization()
+    add_data!(v, x = a[1:end-1], x2=a[2:end],
+        y = b/total,y2=zeros(length(b)))
+    default_scales!(v; typeXaxis = "linear")
+    default_axes!(v)
+
+    #Get unique table name
+    table = v.data[1].name
+
+    #old add_rects code
+    res = VegaMark(_type = "rect",
+        from = VegaMarkFrom(data=table),
+        properties = VegaMarkProperties(enter =
+            VegaMarkPropertySet(x = VegaValueRef(scale = "x", field = "x"),
+                y = VegaValueRef(scale = "y", field = "y"))))
+
+    v.marks == nothing? v.marks = [res] : push!(v.marks, res)
+    #end add_rects code
 
-    v = barplot(x = a[2:end], y = b/total)
+    v.marks[1].properties.enter.x2 = VegaValueRef(scale = "x", field = "x2")
+    v.marks[1].properties.enter.y = VegaValueRef(scale = "y", field = "y")
+    v.marks[1].properties.enter.y2 = VegaValueRef(scale = "y", field = "y2")
+    #v.marks[1].properties.enter.fill = VegaValueRef(scale = "group",
+    #    field = "group")
+    v.marks[1].properties.enter.stroke = VegaValueRef(value="black")
+    v.marks[1].properties.enter.strokeWidth = VegaValueRef(value=0.25)
 
-    #Remove space between bars per theory
-    v.marks[1].properties.enter.width.offset = 0
+    v.axes[findfirst([z._type == "x" for z in v.axes])].title = "x"
+    v.axes[findfirst([z._type == "y" for z in v.axes])].title =
+        relativefreq ? "density" : "counts"
 
+
+    #Return horizontal bar chart
     if horizontal
-        coord_flip!(v)
-        v.marks[1].properties.enter.height.offset = 0
+        #Switch scales
```

Comment:
OK. I think for now, it's good to leave since its the only chart that needs the logic. But since there is no concept of a private function in Julia, having coord_flip!() work for histograms (or any other linear axis) would be the most complete.